desci-server/src/services/Doi.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  PdfComponent,
+  ResearchObjectComponentDocumentSubtype,
+  ResearchObjectComponentType,
+} from '@desci-labs/desci-models';
+import { PrismaClient } from '@prisma/client';
+import { v4 } from 'uuid';
+
+import { DuplicateMintError, NoManuscriptError, BadManifestError, AttestationsError } from '../core/doi/error.js';
+import { logger } from '../logger.js';
+import { IndexedResearchObject, getIndexedResearchObjects } from '../theGraph.js';
+import { ensureUuidEndsWithDot, hexToCid } from '../utils.js';
+
+import { attestationService } from './Attestation.js';
+import { communityService } from './Communities.js';
+import { getManifestByCid } from './data/processing.js';
+
+export class DoiService {
+  dbClient: PrismaClient;
+
+  constructor(prismaClient: PrismaClient) {
+    this.dbClient = prismaClient;
+  }
+
+  async checkMintability(nodeUuid: string) {
```

Comment:
I'd personally split this function into some more granular helpers to aid reading :eyeglasses:

Reply:
E.g., the main function reading as a sequence like below would make understanding the flow super quick imo. 
```ts
assertFirstDoi()
verifyManuscriptIncluded()
validateMetadata()
assertRequiredAttestations()
```

Just a suggestion :open_hands:

Reply:
Totally makes sense

Reply:
Agree